app/views/components/input_date.scala.html

```diff
@@ -14,48 +14,47 @@
  * limitations under the License.
  *@
 
-@import utils.DateErrorFormatter._
+@import viewmodels.Link
 
 @(
-    field: Field,
-    legend: String,
-    legendClass: Option[String] = None,
-    hint: Option[String] = None,
-    headingIsLegend: Boolean = false
+field: Field,
+legend: String,
+legendClass: String = "",
+hint: Option[String] = None,
+link : Option[Link] = None,
+headingIsLegend: Boolean = false,
+extraText: Option[String] = None
 )(implicit messages: Messages)
 
-<div class="form-group @if(field.hasErrors){form-group-error}">
-    <fieldset id="@{field.id}" role="group"
-        @if(hint.nonEmpty || field.hasErrors) {
-            aria-describedby="@if(hint.nonEmpty){hint-@{field.id}}@if(hint.nonEmpty && field.hasErrors){ }@if(field.hasErrors){error-message-@{field.id}-input}"
-        }
-    >
-
-        @if(headingIsLegend){
-            <legend>
-                <h1 @if(legendClass.nonEmpty){class="@legendClass" }else{class="heading-large"}>@legend</h1>
-            </legend>
-        } else {
-            <legend @if(legendClass.nonEmpty){class="@legendClass" }>
-                @legend
-            </legend>
-        }
-        @if(hint.nonEmpty){
-            <div class="form-hint" id="hint-@{field.id}">@hint</div>
-        }
-        @field.errors.map { error =>
-            <span class="error-message" id="error-message-@field.id-input">
-                <span class="visually-hidden">@messages("site.error") </span>@messages(error.message, formatArgs(error.args): _*)
```

Comment:
@andrew-armstrong80 we need the DateErrorFormatter to format date error messages in the correct language